### PR TITLE
Remove Match#anyMatch

### PR DESCRIPTION
**Pull request by RoiEXLab**
This PR is part 4 of a series where usages of the Match class will slowly be replaced by the underlying Predicate.

### CODE REVIEW

File: src/main/java/games/strategy/triplea/ai/AIUtils.java [hunk 1]
```diff
@@ -131,7 +131,8 @@ public static int getIndexOfLastUnitMatching(final List<Unit> units, final Match
   }
 
   static List<Unit> interleaveCarriersAndPlanes(final List<Unit> units, final int planesThatDontNeedToLand) {
-    if (!(Match.anyMatch(units, Matches.unitIsCarrier()) && Match.anyMatch(units, Matches.unitCanLandOnCarrier()))) {
+    if (units.stream().noneMatch(Matches.unitIsCarrier())
+        || units.stream().noneMatch(Matches.unitCanLandOnCarrier())) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ai/AIUtils.java]: Just verifying for my own benefit that this expression still evaluates to `true` when `units` is empty.

File: src/main/java/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java [hunk 2]
```diff
@@ -232,7 +232,8 @@ public static int getUnusedCarrierCapacity(final PlayerID player, final Territor
 
   public static List<Unit> interleaveUnitsCarriersAndPlanes(final List<Unit> units,
       final int planesThatDontNeedToLand) {
-    if (!(Match.anyMatch(units, Matches.unitIsCarrier()) && Match.anyMatch(units, Matches.unitCanLandOnCarrier()))) {
+    if (units.stream().noneMatch(Matches.unitIsCarrier())
+        || units.stream().noneMatch(Matches.unitCanLandOnCarrier())) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java]: Just verifying for my own benefit that this expression still evaluates to `true` when `units` is empty.

File: src/main/java/games/strategy/triplea/delegate/AirBattle.java [hunk 3]
```diff
@@ -117,11 +117,10 @@ private void pushFightLoopOnStack(final boolean firstRun) {
...
 
   private boolean shouldFightAirBattle() {
-    if (m_isBombingRun) {
-      return Match.anyMatch(m_attackingUnits, Matches.unitIsStrategicBomber()) && !m_defendingUnits.isEmpty();
-    } else {
-      return !m_attackingUnits.isEmpty() && !m_defendingUnits.isEmpty();
-    }
+    return ((m_isBombingRun && m_attackingUnits.stream().anyMatch(Matches.unitIsStrategicBomber()))
+        || !m_attackingUnits.isEmpty())
+        && !m_defendingUnits.isEmpty();
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/AirBattle.java]: This is a bit hard to read.

Comment (RoiEXLab) [reply src/main/java/games/strategy/triplea/delegate/AirBattle.java]: I changed the order to be able to see the encapsulation better as part of the rebase, should be better now

**Pull request merged**